lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -364,182 +350,322 @@ _decodeSignedTx bytes = do
+    --   ]
+    sizeOf_Input
+        = sizeOf_SmallArray
+        + sizeOf_Hash32
+        + sizeOf_UInt
+
+    -- transaction_output =
+    --   [address, amount : coin]
+    sizeOf_Output (TxOut addr c)
+        = sizeOf_SmallArray
+        + sizeOf_Address addr
+        + sizeOf_Coin c
+
+    sizeOf_ChangeOutput c
+        = sizeOf_SmallArray
+        + sizeOf_ChangeAddress
+        + sizeOf_Coin c
+
+    -- stake_registration =
+    --   (0, stake_credential)
+    sizeOf_StakeRegistration
+        = sizeOf_SmallArray
+        + sizeOf_SmallUInt
+        + sizeOf_StakeCredential
+
+    -- stake_deregistration =
+    --   (1, stake_credential)
+    sizeOf_StakeDeregistration
+        = sizeOf_StakeRegistration
+
+    -- stake_delegation =
+    --   (2, stake_credential, pool_keyhash)
+    sizeOf_StakeDelegation
+        = sizeOf_SmallArray
+        + sizeOf_SmallUInt
+        + sizeOf_StakeCredential
+        + sizeOf_Hash28
+
+    -- stake_credential =
+    --   [  0, addr_keyhash
+    --   // 1, scripthash
+    --   ]
+    sizeOf_StakeCredential
+        = sizeOf_SmallArray
+        + sizeOf_SmallUInt
+        + sizeOf_Hash28
+
+    -- We carry addresses already serialized, so it's a matter of measuring.
+    sizeOf_Address addr
+        = 2 + toInteger (BS.length (unAddress addr))
+
+    -- For change address, we consider the worst-case scenario based on the
+    -- given wallet scheme. Byron addresses are larger.
+    --
+    -- NOTE: we could do slightly better if we wanted to for Byron addresses and
+    -- discriminate based on the network as well since testnet addresses are
+    -- larger than mainnet ones. But meh.
+    sizeOf_ChangeAddress
+        = case witTag of
+            TxWitnessByronUTxO{} -> 85
+            TxWitnessShelleyUTxO -> 59
+
+    -- Coins can really vary so it's very punishing to always assign them the
+    -- upper bound. They will typically be between 3 and 9 bytes (only 6 bytes
+    -- difference, but on 20+ outputs, one starts feeling it).
+    --
+    -- So, for outputs, since we have the values, we can compute it accurately.
+    sizeOf_Coin
+        = toInteger . BS.length . CBOR.toStrictByteString . CBOR.encodeWord64 . getCoin
+
+    -- withdrawals =
+    --   { * reward_account => coin }
+    sizeOf_Withdrawal
+        = sizeOf_Hash28
+        + sizeOf_LargeUInt
+
+    -- transaction_witness_set =
+    --   { ?0 => [* vkeywitness ]
+    --   , ?1 => [* multisig_script ]
+    --   , ?2 => [* bootstrap_witness ]
+    --   }
+    sizeOf_WitnessSet
+        = sizeOf_SmallMap
+        + sizeOf_VKeyWitnesses
+        + sizeOf_MultisigScript
+        + sizeOf_BootstrapWitnesses
+      where
+        -- ?0 => [* vkeywitness ]
+        sizeOf_VKeyWitnesses
+            = (if numberOf_VkeyWitnesses > 0 then sizeOf_Array + sizeOf_SmallUInt else 0)
+            + sizeOf_VKeyWitness * numberOf_VkeyWitnesses
+
+        -- ?1 => [* multisig_script ]
+        sizeOf_MultisigScript
+            = 0
```

Comment:
I bet this one is to be updated very soon

Reply:
Yes. I was thinking about adding a `TODO` tag with a ticket number maybe ?